### PR TITLE
add dev workflow file

### PR DESCRIPTION
**Pull request by YoninL**

### CODE REVIEW

File: .github/workflows/build&deploy-prod.yml [hunk 1]
```diff
@@ -2,10 +2,8 @@ name: Shipping to AWS ECS from Source Code
   
 on:
   workflow_dispatch:
-  push:
-    branches: [ "main" ]
-  pull_request:
-    branches: [ "main" ]
+  tags:
```
Comment (rickliujh) [inline .github/workflows/build&deploy-prod.yml]: is it messing push in line 5

File: .github/workflows/build&deploy-dev.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
+name: Shipping to AWS ECS from Source Code
+  
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main" ]
```
Comment (rickliujh) [inline .github/workflows/build&deploy-dev.yml]: is deploying when push to main branch necessary

File: .github/workflows/build&deploy-dev.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  
+
+env:
+  APP_ENV: dev
+  AWS_REGION: us-west-2
+  ECR_REPOSITORY: billing-and-payment-service
+  ECS_SERVICE: billing-and-payment-service
+  ECS_CLUSTER: flipgod-dev
+  ECS_TASK_DEFINITION: billing-and-payment-service-dev/ecs/task-definition.json
```
Comment (rickliujh) [inline .github/workflows/build&deploy-dev.yml]:
```suggestion
  ECS_TASK_DEFINITION: billing-and-payment-service/ecs/dev/task-definition.json
```

File: .github/workflows/build&deploy-dev.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  
+
+env:
+  APP_ENV: dev
+  AWS_REGION: us-west-2
+  ECR_REPOSITORY: billing-and-payment-service
```
Comment (rickliujh) [inline .github/workflows/build&deploy-dev.yml]: needs an ecr repo for dev env called dev-billing-and-payment-service?

File: .github/workflows/build&deploy-dev.yml (hunk 3 above)
Comment (YoninL) [reply .github/workflows/build&deploy-dev.yml]: ok, will change to ecs/dev

File: .github/workflows/build&deploy-dev.yml (hunk 4 above)
Comment (YoninL) [reply .github/workflows/build&deploy-dev.yml]: I will add "-dev" to the image tag, yes, we can use a different ecr repo as well. What's your choice?

File: .github/workflows/build&deploy-prod.yml (hunk 1 above)
Comment (YoninL) [reply .github/workflows/build&deploy-prod.yml]: fixed.

File: .github/workflows/build&deploy-dev.yml (hunk 2 above)
Comment (YoninL) [reply .github/workflows/build&deploy-dev.yml]: updated.

**Comment by YoninL** [comment]
bps on dev is now good, please approve and merge.

**Pull request merged**